airflow/plugins/operators/airtable_to_warehouse.py

```diff
@@ -0,0 +1,124 @@
+import pandas as pd
+import os
+
+from pyairtable import Table
+from calitp import save_to_gcfs, to_snakecase, write_table
+
+from airflow.models import BaseOperator
+
+
+def airtable_to_df(
+    air_base_id,
+    air_table_name,
+    id_name="__id",
+    rename_fields=None,
+    column_prefix=None,
+    api_key=None,
+):
+    """Download an airtable as a DataFrame.
+
+    Note that airtable records have rows structured as follows:
+        [{"id", "fields": {colname: value, ...}, ...]
+
+    This function applies renames in the following order.
+
+        1. rename id
+        2. rename fields
+        3. apply column prefix (to columns not renamed by 1 or 2)
+    """
+
+    api_key = os.environ["CALITP_AIRTABLE_API_KEY"] if not api_key else api_key
+
+    if rename_fields:
+        if not isinstance(rename_fields, dict):
+            raise TypeError("rename fields must be a dictionary")
+    else:
+        rename_fields = {}
+
+    print(f"Downloading airtable data for {air_base_id}.{air_table_name}")
+    all_rows = Table(api_key, air_base_id, air_table_name).all()
+    raw_df = pd.DataFrame([{id_name: row["id"], **row["fields"]} for row in all_rows])
+
+    # rename fields follows format new_name: old_name
+    final_df = raw_df.rename(columns={v: k for k, v in rename_fields.items()}).pipe(
+        to_snakecase
+    )
+
+    if column_prefix:
+        return final_df.rename(
+            columns=lambda s: s
+            if (s in rename_fields or s == id_name)
+            else f"{column_prefix}{s}"
+        )
+
+    return final_df
+
+
+class AirtableToWarehouseOperator(BaseOperator):
+    def __init__(
+        self,
+        air_base_id,
+        air_table_name,
+        table_name,
+        id_name="__id",
+        gcs_path=None,
+        rename_fields=None,
+        column_prefix=None,
+        api_key=None,
+        **kwargs,
+    ):
+        """An operator that will download data from an airtable and load it into
+            BigQuery and/or save it as a CSV file with the current date in Google Cloud
+            Storage (GCS).
+
+        Args:
+            air_base_id (str): The underlying id of the airtable instance being used.
+            air_table_name (str): The table name that should be extracted from the
+                airtable base
+            table_name (str): The name of the table to save to in Big Query
+            id_name (str, optional): The name to give the ID column. Defaults to "__id".
+            gcs_path (str, optional): A GCS path prefix. If not provided, no uploading
+                to GCS will occur. The resulting item gets saved to GCS at the following
+                path:
+                    `{base_calitp_bucket}/{gcs_path}/{execution_date}/{table_name}.csv`.
+            rename_fields (list, optional): A list of objects where each object consists
+                of a string key and value representing the new desired column name and
+                airtable column name respectively. Defaults to None.
```

Comment:
**issue:** Is this correct? Seems like it's a `dict` above in `airtable_to_df`.

```suggestion
            rename_fields (dict, optional): A mapping new desired column name (string) to
                current airtable column name (string) respectively. Defaults to None.
```

**nitpick:** On a related note, why are we using `new_name: old_name`? I feel like I've more commonly seen `old_name: new_name`, and would probably expect that in the absence of instructions. I don't think I have a strong preference about it, but am curious why this order.

Reply:
Yeah, you're right. I went ahead and made my own commit (1d27b283d0e5b091ad3d2a0e0b20a65c649c2b7e) to improve your wording a bit.